src/docfx/template/TemplateEngine.cs

```diff
@@ -29,11 +30,27 @@ public TemplateEngine(string templateDir, string locale)
             Global = LoadGlobalTokens(templateDir, _locale);
         }
 
+        public static TemplateEngine Create(Docset docset)
+        {
+            Debug.Assert(docset != null);
+
+            if (string.IsNullOrEmpty(docset.Config.Theme))
+            {
+                return null;
+            }
+
+            var (themeRemote, themeBranch) = LocalizationUtility.GetLocalizedTheme(docset.Config.Theme, docset.Locale, docset.Config.Localization.DefaultLocale);
+            var (themePath, themeLock) = docset.RestoreMap.GetGitRestorePath($"{themeRemote}#{themeBranch}", docset.DependencyLock);
+            Log.Write($"Using theme '{themeRemote}#{themeLock.Commit}' at '{themePath}'");
+
+            return new TemplateEngine(themePath, docset.Locale);
+        }
+
         public string Render(PageModel model, Document file)
         {
             // TODO: only works for conceptual
             var content = model.Content.ToString();
-            var (templateModel, metadata) = TemplateTransform.Transform(model, file);
+            var (templateModel, metadata) = TemplateTransform.Transform(this, model, file);
```

Comment:
seems a bit odd to pass `this` here, should `TemplateTransform` be part of `TemplateEngine`?

Reply:
i am going to leave this to another PR or discussion, @live1206  @yufeih